velox/core/SimpleFunctionMetadata.h

```diff
@@ -124,6 +124,54 @@ struct TypeAnalysisResults {
   void resetTypeString() {
     out.str(std::string());
   }
+
+  // Set a priority based on the collected information. Lower priorities are
+  // picked first during function resolution. Each signature get a rank out
+  // of 4, those ranks form a Lattice ordering.
+  // rank 1: generic free and variadic free.
+  //    e.g: int, int, int -> int.
+  // rank 2: has variadic but generic free.
+  //    e.g: Variadic<int> -> int.
+  // rank 3: has generic but no variadic of generic.
+  //    e.g: Generic<>, Generic<>, -> int.
+  // rank 4: has variadic of generic.
+  //    e.g: Variadic<Generic<>> -> int.
+
+  // If two functions have the same rank, then concreteCount is used to
+  // to resolve the ordering.
+  // e.g: consider the two functions:
+  //    1. int, Generic<>, Variadic<int> -> has rank 3. concreteCount =2
+  //    2. int, Generic<>, Generic<>     -> has rank 3. concreteCount =1
+  // in this case (2) is picked.
```

Comment:
Wouldn't (1) be picked since it has more concrete types?

Reply:
yeh 1 will fix the comment.